src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,168 @@
+     * 
+     * @return an iterable of RecommendationRequest
+     */
+    @Operation(summary= "List all recommendation requests")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<RecommendationRequest> allRecommendationRequests() {
+        Iterable<RecommendationRequest> requests = recommendationRequestRepository.findAll();
+        return requests;
+    }
+
+    /**
+     * Create a new request
+     * 
+     * @param requesterEmail   
+     * @param requesterName     
+     * @param professorEmail  
+     * @param professorName
+     * @param requestType      
+     * @param details           
+     * @param submissionDate
+     * @param completionDate
+     * @param status
+     * @return                  
+     */
+    @Operation(summary= "Create a new request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public RecommendationRequest postRecommendationRequest(
+            @Parameter(name="requesterEmail") @RequestParam String requesterEmail,
+            @Parameter(name="requesterName") @RequestParam String requesterName,
+            @Parameter(name="professorEmail") @RequestParam String professorEmail,
+            @Parameter(name="professorName") @RequestParam String professorName,
+            @Parameter(name="requestType") @RequestParam String requestType,
+            @Parameter(name="details") @RequestParam String details,
+            @Parameter(name="submissionDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("submissionDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime submissionDate,
+            @Parameter(name="completionDate", description="date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("completionDate") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime completionDate,
+            @Parameter(name="status") @RequestParam String status)
+            throws JsonProcessingException {
+
+        // For an explanation of @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+        // See: https://www.baeldung.com/spring-date-parameters
+
+        // log.info("dateRequested={}", dateRequested);
+        // log.info("dateNeeded={}", dateNeeded);
+
+        RecommendationRequest recommendationRequest = new RecommendationRequest();
+        recommendationRequest.setRequesterEmail(requesterEmail);
+        recommendationRequest.setRequesterName(requesterName);
+        recommendationRequest.setProfessorEmail(professorEmail);
+        recommendationRequest.setProfessorName(professorName);
+        recommendationRequest.setRequestType(requestType);
+        recommendationRequest.setDetails(details);
+        recommendationRequest.setSubmissionDate(submissionDate);
+        recommendationRequest.setCompletionDate(completionDate);
+        recommendationRequest.setStatus(status);
```

Comment:
Status should be hard coded as `pending` on a newly created request.  The end user doesn't get to "choose" what status a new request is in.  It is *always* in the pending state, and only the professor gets to change it to either `accepted` or `denied`